src/components/results/ResultsTabs.vue

```diff
@@ -0,0 +1,127 @@
+<script lang="ts">
+/**
+ * ResultsTabs Component
+ *
+ * @description * Use this component to render tabs on the ResultsDetailPage.
+ *
+ * @components
+ * - `ResultsGrid`: Component to render a data report in a grid.
+ * - `ResultsTable`: Component to render rankings in a table.`
+ *
+ * @example
+ * <results-tabs />
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858%3A105197&t=4cALO2fsjKI90AW1-1)
+ */
+
+// libraries
+import { defineComponent, ref } from 'vue';
+
+// routes
+import { routesConf } from 'src/router/routes_conf';
+
+type ResultsTabsOption = 'report' | 'regularity' | 'performance';
+
+export default defineComponent({
+  name: 'ResultsTabs',
+  props: {
+    locked: {
+      type: Array as () => ResultsTabsOption[],
+      default: () => [],
+    },
+  },
+  setup(props) {
+    // list of available tabs
+    const tabs: ResultsTabsOption[] = ['report', 'regularity', 'performance'];
+    const activeTab = ref('' as ResultsTabsOption);
+
+    // locked tabs - exposed for testing and further logic
+    const lockedTabs = props.locked;
+
+    /**
+     * Checks if the given tab is locked.
+     * @param tab the tab to check
+     */
+    const isLocked = (tab: ResultsTabsOption): boolean => {
+      if (!lockedTabs.length) return false;
```

Comment:
What about [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator)?

Reply:
Updated.
Generally, I tend to use early returns for better readability, but in a simple function, I agree this is good.